src/jsCallback.php

```diff
@@ -53,44 +53,60 @@ public function set($callback, $args = [])
         }
 
         return parent::set(function () use ($callback) {
-            $chain = new jQuery(new jsExpression('this'));
+            try {
+                $chain = new jQuery(new jsExpression('this'));
 
-            $values = [];
-            foreach ($this->args as $key=>$value) {
-                $values[] = isset($_POST[$key]) ? $_POST[$key] : null;
-            }
+                $values = [];
+                foreach ($this->args as $key=>$value) {
+                    $values[] = isset($_POST[$key]) ? $_POST[$key] : null;
+                }
 
-            $response = call_user_func_array($callback, array_merge([$chain], $values));
+                $response = call_user_func_array($callback, array_merge([$chain], $values));
 
-            if ($response === $chain) {
-                $response = null;
-            }
+                if ($response === $chain) {
+                    $response = null;
+                }
 
-            $actions = [];
+                $actions = [];
 
-            if ($chain->_chain) {
-                $actions[] = $chain;
-            }
+                if ($chain->_chain) {
+                    $actions[] = $chain;
+                }
 
-            $response = $this->flatternArray($response);
-
-            foreach ($response as $r) {
-                if (is_string($r)) {
-                    $actions[] = new jsExpression('alert([])', [$r]);
-                } elseif ($r instanceof jsExpressionable) {
-                    $actions[] = $r;
-                } elseif ($r === null) {
-                    continue;
-                } else {
-                    throw new Exception(['Incorrect callback. Must be string or action.', 'r'=>$r]);
+                $response = $this->flatternArray($response);
+
+                foreach ($response as $r) {
+                    if (is_string($r)) {
+                        $actions[] = new jsExpression('alert([])', [$r]);
+                    } elseif ($r instanceof jsExpressionable) {
+                        $actions[] = $r;
+                    } elseif ($r === null) {
+                        continue;
+                    } else {
+                        throw new Exception(['Incorrect callback. Must be string or action.', 'r'=>$r]);
+                    }
                 }
-            }
 
-            $ajaxec = implode(";\n", array_map(function (jsExpressionable $r) {
-                return $r->jsRender();
-            }, $actions));
+                $ajaxec = implode(";\n", array_map(function (jsExpressionable $r) {
+                    return $r->jsRender();
+                }, $actions));
+
+                $this->app->terminate(json_encode(['success'=>true, 'message'=>'Success', 'eval'=>$ajaxec]));
+            } catch (\atk4\data\ValidationException $e) {
```

Comment:
1. As Alain said in his comment - not sure why we need message and eval=alert both at the same time.

2. I still don't know how to run UI JS on my local repo. JS methods simply return json and nothing catches it.
See video: https://www.useloom.com/share/c81f35682c5648ea8f1885fd5736c4dc